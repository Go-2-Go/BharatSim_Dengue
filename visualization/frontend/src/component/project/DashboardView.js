import React from 'react';
import Box from '@material-ui/core/Box';

function TabPanel(props) {
  const { children, value, index } = props;

  return <div>{value === index && <Box p={3}>{children}</Box>}</div>;
}

export default function DashboardView({ views, value }) {
  return (
    <Box>
      {views.map((view, index) => (
        <TabPanel value={value} index={index} key={`value-${view.name}`}>
          {view.name}
        </TabPanel>
      ))}
    </Box>
  );
}
