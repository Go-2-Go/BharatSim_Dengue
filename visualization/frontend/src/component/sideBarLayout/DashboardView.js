import React from 'react';
import Box from '@material-ui/core/Box';
import TabPanel from '../../uiComponent/TabPanel';
import ConfigureDashboardData from '../project/ConfigureDashboardData';

export default function DashboardView({ projectName, dashboardData, value }) {
  return (
    <Box>
      {dashboardData.map((dashboard, index) => (
        <TabPanel value={value} index={index} key={`value-${dashboard.name}`}>
          <ConfigureDashboardData dashboardData={dashboard} projectName={projectName} />
        </TabPanel>
      ))}
    </Box>
  );
}
