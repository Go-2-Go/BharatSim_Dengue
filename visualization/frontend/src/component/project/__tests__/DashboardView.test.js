import React from 'react';
import { render } from '@testing-library/react';
import DashboardView from '../DashboardView';

const views = [{ name: 'dashbaord1' }, { name: 'dashbaord2' }];
describe('DashboardView', () => {
  it('should match snapshot for provided value', function () {
    const { container } = render(<DashboardView value={1} views={views} />);
    expect(container).toMatchSnapshot();
  });
});
