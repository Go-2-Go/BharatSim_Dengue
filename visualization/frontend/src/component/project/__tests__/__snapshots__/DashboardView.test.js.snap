// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardView should match snapshot for provided value 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-1"
  >
    <div />
    <div>
      <div
        class="MuiBox-root MuiBox-root-2"
      >
        dashbaord2
      </div>
    </div>
  </div>
</div>
`;
